libs/ledgerjs/packages/hw-app-exchange/src/ReturnCode.ts

```diff
@@ -32,7 +32,10 @@ export function getExchangeErrorMessage(errorCode: number): string | undefined {
     case ErrorStatus.INVALID_ADDRESS:
       return "Invalid address";
     case ErrorStatus.USER_REFUSED:
-      return "User refused";
+      if (step === "CHECK_PAYOUT_ADDRESS")
+        return "This receiving account does not belong to your device. Please change and retry."
+      else if (step === "CHECK_REFUND_ADDRESS")
+        return "This receiving account does not belong to your device for the refund. Please change and retry.";
```

Comment:
Shouldn't you fallback to the previous error? And this step checks was on INVALID_ADDRESS, not USER_REFUSED iirc?